src/util.hpp

```diff
@@ -1,17 +1,19 @@
 #pragma once
-
 #include <filesystem>
 #include <fstream>
 #include <iomanip>
 #include <iostream>
 #include <sstream>
 #include <string>
+#include <unordered_map>
 #include <vector>
 
 namespace util {
 // return vector of strings containing each line of the given file
 std::vector<std::string> get_lines(std::filesystem::path const &filename);
 
-// returns a vector of words contained in a string
-// std::vector<std::string> separate_substrings(const std::string &str);
+// extract key-value pairs from given vector of strings and insert them in an
+// unordered_map
+std::unordered_map<std::string, std::string>
```

Comment:
Weird lines breaks, I suggest you add a `clang-format` file to clean up your formatting everywhere.